newsrevenuehub/rev-engine
# DEV-2005 | unable to request magic link for contributor portal when domains are gt 20 characters long

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

#### What's this PR do?

- Renames `SLUG_MAX_LENGTH` to `RP_SLUG_MAX_LENGTH` so meaning aligns with use
- Removes max length limit on  `ContributorObtainTokenSerializer.subdomain`.  This was originally set at 20 characters and was the cause of the bug with magic link email requests. 
- Removes max length limit  on `ContributionMetadataSerializer.revenue_program_slug`. This code was already touched by renaming `SLUG_MAX_LENGTH` TO `RP_SLUG_MAX_LENGTH`, and I concluded that there shouldn't be a limit here in the first place.


The reason for not limiting subdomain in `ContributorObtainTokenSerializer.subdomain` has to do with how that serializer is used. It is narrowly used by the SPA to request magic email links. The SPA derives the current subdomain from `window.location.host`, and passes that value along in the body of the request to `RequestContributorTokenEmailView`. Thus if a request gets made where the value for subdomain is beyond the max length enforced at ORM layer (which is set to `RP_SLUG_MAX_LENGTH`, either a malicious actor is interacting directly with the API layer, or the SPA is somehow being served from a subdomain that violates the constraints around subdomain/rp-slug length. In neither case does it make sense to send validation errors about max length of subdomain request body param.

The reason for not limiting `ContributionMetadataSerializer.revenue_program_slug` is similar. That serializer gets used in the view that the SPA posts to when a user submits a stripe payment. The value that gets sent for `revenue_program_slug` is derived in the same way from `window.location.host`.

#### Why are we doing this? How does it help us?

Makes form for requesting magic link emails work when the subdomain is greater than 20 characters.


#### How should this be manually tested?

Go to https://newsrevenuehub-dev-2005.revengine-review.org/contributor and submit the form with a valid email address. It should submit, and you get the normal message about an email being sent.

#### Have automated unit tests been added?

No. Changes in this PR remove business logic in serializer layer around enforcing max length on two fields. That business logic arguably should have been tested in past, but since it's being removed, there's no need for additional tests.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

N/A

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-2005

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No

#### Are there next steps? If so, what? Have tickets been opened for them?

Sort of. Decoupled from this ticket, but the SPA should be refactored to display meaningful feedback to user when a request for magic email link fails. That's tracked in [DEV-2020](https://news-revenue-hub.atlassian.net/browse/DEV-2020)
